test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -447,10 +458,142 @@ TEST_F(HttpConnectionManagerConfigTest, ReversedEncodeOrderConfig) {
+  const std::string config_yaml = R"EOF(
+route_config:
+  name: foo_route
+scoped_routes_config:
+  name: foo_route_scopes
+codec_type: auto
+stat_prefix: foo
+http_filters:
+  - name: envoy.router
+)EOF";
+
+  EXPECT_THROW_WITH_MESSAGE(
+      HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(config_yaml), context_,
+                                  date_provider_, route_config_provider_manager_,
+                                  scoped_routes_config_provider_manager_),
+      EnvoyException, "Error: RDS must be used when scoped routing is enabled");
+
+  const std::string config_yaml2 = R"EOF(
+route_config:
+  name: foo_route
+scoped_rds:
+  config_source: { path: /path/to/file }
+  scoped_routes_config_set_name: foo_scope
+codec_type: auto
+stat_prefix: foo
+http_filters:
+  - name: envoy.router
+)EOF";
+
+  EXPECT_THROW_WITH_MESSAGE(
+      HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(config_yaml2), context_,
+                                  date_provider_, route_config_provider_manager_,
+                                  scoped_routes_config_provider_manager_),
+      EnvoyException, "Error: RDS must be used when scoped routing is enabled");
+
+  const std::string config_yaml3 = R"EOF(
+rds:
+  config_source: { path: /path/to/file }
+  route_config_name: foo_routes
+scoped_routes_config:
+  name: foo_route_scopes
+codec_type: auto
+stat_prefix: foo
+http_filters:
+  - name: envoy.router
+)EOF";
+
+  EXPECT_THROW_WITH_MESSAGE(
+      HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(config_yaml3), context_,
+                                  date_provider_, route_config_provider_manager_,
+                                  scoped_routes_config_provider_manager_),
+      EnvoyException,
+      "Error: the RDS subscription specifier must be set to scoped_rds_template=true "
+      "when scoped routing is enabled");
+
+  const std::string config_yaml4 = R"EOF(
+rds:
+  config_source: { path: /path/to/file }
+  scoped_rds_template: true
+codec_type: auto
+stat_prefix: foo
+http_filters:
+  - name: envoy.router
+)EOF";
+
+  EXPECT_THROW_WITH_MESSAGE(
+      HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(config_yaml4), context_,
+                                  date_provider_, route_config_provider_manager_,
+                                  scoped_routes_config_provider_manager_),
+      EnvoyException,
+      "Error: RDS must specify a route_config_name when scoped routing is not enabled");
+}
+
+TEST_F(HttpConnectionManagerConfigTest, ScopedRoutingValidationSuccess) {
+  const std::string config_yaml = R"EOF(
+rds:
+  config_source: { path: /path/to/file }
+  scoped_rds_template: true
+scoped_routes_config:
+  name: foo_route_scopes
+codec_type: auto
+stat_prefix: foo
+http_filters:
+  - name: envoy.router
+)EOF";
+
+  HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(config_yaml), context_,
```

Comment:
use `EXPECT_NO_THROW` to indicate that we don't expect this to throw

Reply:
Done.